include/aspect/utilities.h

```diff
@@ -838,6 +838,41 @@ namespace aspect
                               const SymmetricTensor<2,dim> &dviscosities_dstrain_rate,
                               const double safety_factor);
 
+
+    /**
+     * This function computes the weighted average $\bar y$ of $y_i$  for a weighted p norm of which
+     * the sum weighs are assumed to be one. This leads for a general p to:
+     * $\bar y = \left(\frac{\sum_{i=1}^k w_i y_i^p}{\sum_{i=1}^k w_i}\right)^{\frac{1}{p}}$.
+     * When p = 0 we take the geometric average:
+     * $\bar y = \exp\left(\frac{\sum_{i=1}^k w_i \log\left(y_i\right)}{\sum_{i=1}^k w_i}\right)$,
```

Comment:
above you say that the weights are supposed to sum to one. is this the case only for "general p"?